pandas/src/algos_common_helper.pxi.in

```diff
@@ -83,8 +83,8 @@ def pad_{{name}}(ndarray[{{c_type}}] old, ndarray[{{c_type}}] new,
     if limit is None:
         lim = nright
     else:
-        if limit < 0:
-            raise ValueError('Limit must be non-negative')
+        if limit < 1:
```

Comment:
so here as well, something like
if not util.is_integer_object(limit):
    raise ValueError('limit must be an integer')
if limit < 1:
    raise ValueError(....)
```